
---
title: "Summarize vcf data in few functions"
author: Marika Oksanen
date: "'r Sys.Date()'"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{RaukRGenome_vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(RaukRGenome)
```

This package can be used to get summary numbers of a vcf file, including what variants 
are included and how many of them. It is also possible to count how many times 
different variant types occur in the genotype data of the vcf file. 

