---
title: "Project 1: Genomic Data"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Calculating genetic differentiation

```{r, echo = F}
suppressMessages(require(ExploreData))
suppressMessages(require(vcfR))
suppressMessages(require(tidyverse))
suppressMessages(require(reshape2))
suppressMessages(require(dplyr))
suppressMessages(require(tidyr))
suppressMessages(require(adegenet))
suppressMessages(require(hierfstat))
suppressMessages(require(pegas))
suppressMessages(require(mmod))
suppressMessages(require(ggplot2))
```

Loading the data
```{r, echo = F}
system.file("extdata", "chr22.1000g.vcf.tar.gz", package="ExploreData", mustWork = T)
```

## Loading the data

```{r}
vcf <- read.vcfR( "/Library/Frameworks/R.framework/Versions/4.0/Resources/library/ExploreData/extdata/chr22.1000g.vcf")
Population_info <- read.table("Population_info.txt", sep="\t", header=TRUE)
```

## Adjusting the data structure

```{r}
tidy <- vcfR2tidy(vcf)
All_samples <- tidy$gt
All_samples = subset(All_samples, select = -c(ChromKey, gt_GT)) # keeping only the needed columns
All_samples$gt_GT_alleles <- gsub("\\|", "", as.character(All_samples$gt_GT_alleles)) # remove the | from the genotype (G|G)
All_samples_filtered <- subset(All_samples, nchar(as.character(gt_GT_alleles)) <= 2) # keep only the SNPs
All_samples_filtered_unique <- subset(All_samples_filtered, POS != "19512392") # remove the position 19512392, because it gives me troubles
All_samples_reshape <- dcast(All_samples_filtered_unique, Indiv ~ POS, value.var = "gt_GT_alleles") # reshape the table from long structure to the wide structure
names(Population_info)[1] <- "Indiv" # Change the column name so that it matches the other data.frame's column name
Mydata <- merge(Population_info, All_samples_reshape, by = "Indiv", all = T) # merge the genotype data with the population info data
```

## Data conversion

```{r}
Indiv <- as.character(Mydata$Indiv)
Sex <- as.character(Mydata$Sex)
Superpopulation <- as.character(Mydata$Superpopulation_code)
Population <- as.character(Mydata$Population_name)

locus <- Mydata[, 10:19774]
Mydata1 <- df2genind(locus, ploidy = 2, ind.names = Indiv, pop = Superpopulation, sep = "")
```


## Observed and expected heterozygosity: 

```{r}
Basic_stats <- basic.stats(Mydata1)
boxplot(Basic_stats$perloc)
```

```{r}
# set the population info into in Mydata1 strata slot.
strata(Mydata1) <- Population_info
```


```{r}
setPop(Mydata1) <- ~Superpopulation_code 
diff <- diff_stats(Mydata1)
```

```{r}
per.locus <- melt(diff$per.locus, varnames = c("Locus", "Statistic"))
stats     <- c("Hs", "Ht", "Gst", "Gprime_st", "D", "D")
glob      <- data.frame(Statistic = stats, value = diff$global)
```

```{r}
set.seed(20151219) 
reps <- chao_bootstrap(Mydata1, nreps = 10)
summarise_bootstrap(reps, D_Jost) # Using the D_Jost function to summarize.
```

```{r}
distance  <- dist(Mydata1) # Euclidean distance
Strata  <- strata(Mydata1)
Amova <- pegas::amova(distance ~ Superpopulation_code/Population_name, data = Strata, nperm = 0)
```

```{r}
set.seed(20151219)
Amova <- pegas::amova(distance ~ Superpopulation_code/Population_name, data = Strata, nperm = 10)
```

```{r}
setPop(Mydata1) <- ~Superpopulation_code
DAPC <- dapc(Mydata1, n.pca = 100, n.da = 4) # 100 principal components; 4 discriminant functions
scatter(DAPC)
```










