---
title: "RaukR 2021 - Final project"
subtitle: "∝  Genome analyses - Human Chromosome 22  ∝"
author: "Authors: AsEf, MAB, MaOk, Phil, ZLi"
institute: "NBS Training"
date: "2021/06/24" #(updated: `r Sys.Date()`)"
output:
  xaringan::moon_reader:
    css: [default, metropolis , metropolis-fonts]
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
---
class:center, middle
```{r warning=FALSE, include=FALSE, message=FALSE}
require(formatR)
renv::install("tidyverse", "tibble")
devtools::install_github("gadenbuie/countdown")
require(countdown)
countdown(minutes = 10)
```
# **GROUP MEMBERS**

## **Astradeni Efthymiadou**
## **Marika Oksanen**
## **Mónica Angulo-Bedoya**
## **Philip Dusart**
## **Zhuang Liu**

### .right[Advisor: Nina Norgren]     
---

class: center 
# **Project** - Analyses of a VCF format file with population allele frequencies of variants on human chromosome 22. 

## **What is a VCF file? How does it look like? **


![](https://user-images.githubusercontent.com/35312839/34796524-c4ddee7a-f623-11e7-99a3-4b6d193c894f.png)

![]](https://media.tenor.com/images/80a69c16ccffac7c6ec418b3d5e6827f/tenor.gif)
---
class: center

# Different nationalities, cultures, and backgrounds. Therefore different perspectives

.left[
## **Read as a tibble - option 1** - Working with pipelines
```{r,echo=TRUE, include=TRUE, warning = FALSE, message = FALSE}
#renv::install("tidyverse", "tibble")

require(tidyverse)
require(tibble)

tib_vcf = "chr22.1000g.copy.vcf" %>%
  read.table() %>%
  as_tibble() %>%
  setNames(.,c("chrom", "pos", "new-var", "ref", "alt", "phred", 
               "filter", "format", "GT", paste0(rep("ind",2504),"-", 1:2504)))

head(tib_vcf[1:8,1:8])

```
]
---
```{r,echo=TRUE, include=TRUE, warning = FALSE, message = FALSE}
splitcol<-function(x) {
  f<-strsplit(x,"=")
  setNames(sapply(f,'[',2), sapply(f,'[',1))
}
#Making each row into a vector
fmt<-lapply(strsplit(tib_vcf$format,";"), splitcol)

#new column names based on "format"
col_nam<-unique(unlist(sapply(fmt, names)))

#extract data for all rows from each column
row_dat<-do.call(rbind, lapply(fmt, '[', col_nam))

#put together the new data set (format split) with the original dataset
#be careful with the column number, so you introduce this data where you really want to.
vcf_det<-cbind(tib_vcf[,1:7], row_dat[,1:12], tib_vcf[,9:2513])

head(vcf_det[1:8,1:12])

```
---
class: left, middle
## **Read as a tibble - option 2**
# Functions:

```{r class:left, echo=TRUE, message=TRUE, warning=FALSE, include=TRUE}

info_extract <- function(data = NULL, info = NULL){
  #split the string
  tmp_a <- data %>% str_split(., pattern = ";") %>% unlist()
  tmp_b <- lapply(tmp_a, FUN = function(x){
    return(x %>% str_split(., pattern = "=") %>% unlist())
  })
  tmp_c <- do.call('rbind', tmp_b) %>% t() %>% as.data.frame()
  colnames(tmp_c) <- tmp_c[1,]
  tmp_c <- tmp_c[-1,]
  return(tmp_c[info])
}
```

---
class: middle
```{r eval=TRUE, tidy=FALSE, warning=FALSE, message=FALSE}
source("z-project_Functions.R")
library(tidyverse)
library(vroom)
chr22vcf <- readRDS("z-Data/chr22vcf.rds")
colnames(chr22vcf) <- c("chr", "pos", "id", "ref", "alt", "qual", 
                        "filter", "info", "format", paste0("sm", 1:2504))
chr22vcf[1:10,1:15]
```
---
# Some results
## **1. Extract the sequencing info**
### * Extract the SNP info
```{r eval=TRUE, tidy=FALSE, message=F, warning=F}
# extract the info using self-written function, like the VT (Variant Type) is the type of SNP or INDEL
vcf_info_VT <- sapply(chr22vcf$info, FUN = function(x) info_extract(data = x, info = "VT"))
vcf_info_VT <- do.call("rbind", vcf_info_VT) %>% as_tibble()
```
### * Plot the number of different types of variants
```{r fig.width=12, fig.height=8, warning=FALSE, message=FALSE}
require(ggbreak)
vt_plot <- table(vcf_info_VT) %>% as_tibble() %>% 
  rename(VT = vcf_info_VT) %>% 
  mutate(VT=parse_factor(VT, levels = c("SNP", "INDEL", "SNP,INDEL", "SV"))) %>%
  ggplot(., aes(x=VT, y=n, color = VT, fill=VT)) +   geom_bar(stat="identity")  + scale_color_manual(values = c("#66c2a5", "#fc8d62", "#8da0cb", "#e78ac3")) +
  scale_fill_manual(values = c("#66c2a5", "#fc8d62", "#8da0cb", "#e78ac3")) +
  geom_text(aes(label=n), vjust=1.6, color="black", size=4) + 
  labs(x="", y="Frequency", title = "Variant Numbers") +
  theme_classic() + theme(legend.position = "none") +
  scale_y_break(c(20, 700), scales=0.5) +   scale_y_break(c(740,19000), scales=1)
```
---

```{r fig.width=8, fig.height=6, fig.align = 'center'}
vt_plot
```

---
class:left, middle
## **2. Extract and process the population info **
### * Extract populations info

```{r message=F, warning=F}
ori <- readRDS("z-Data/ALL.chr22.origin.rds")
#Just need the sample information
sm_info <- ori[1,] %>% t() %>% as_tibble() %>% slice(-c(1:9))
colnames(sm_info) <- "Sample_name"
all_sminfo <- vroom("z-Data/igsr_samples.tsv")
colnames(all_sminfo) <- gsub(" ","_",colnames(all_sminfo))
#Get the sample info what we need for our project
sm_info <- sm_info %>% 
  left_join(., all_sminfo, by=c("Sample_name" = "Sample_name"))
#delete the temporary files
rm(ori, all_sminfo)
```

---
class: left, middle
## **3. Draw a map of the distribution of samples**
### * Draw the map of population distribution

```{r echo=TRUE, message=F, warning=F}
pop_loc <- vroom("z-Data/igsr_populations.tsv")
colnames(pop_loc) <- gsub(" ","_",colnames(pop_loc))

#add the population latitude and longitude
uni_pop_n <- str_split(sm_info[["Population_name"]], pattern = ",") %>% unlist() %>% sort() %>% table() %>% as_tibble()
colnames(uni_pop_n)[1] <- "Population_name"
uni_pop <- str_split(sm_info[["Population_name"]], pattern = ",") %>% unlist() %>% unique() %>% sort()
sm_loc <- pop_loc %>% select(3:6, 9) %>% 
  filter(Population_name %in% uni_pop) %>% 
  left_join(., uni_pop_n, by=c("Population_name" = "Population_name")) %>% 
  distinct(Population_name, n, .keep_all = TRUE)
```

---
### *Ploting map...
```{r echo=TRUE, message=F, warning=F}
require(ggplot2)
require(maps)
require(ggrepel)
#Get the info of world map
world <- map_data("world")
map_plot <- ggplot() +
  geom_polygon(data = world, aes(x=long, y = lat, group = group, fill=group),  alpha=0.3, show.legend = FALSE) + #fill="grey",
  geom_point(data=sm_loc, aes(x=Population_longitude, y=Population_latitude, color=Population_name, size=n), alpha=1, show.legend = FALSE) +
  scale_size_continuous(range = c(1,4), breaks = c(seq(20,120, by=20))) + theme_void() +   labs(caption = "RaukR 2021 Genomic Data Project: Population Distribution") +   theme(plot.background = element_rect(fill = "white", color = NA), panel.background = element_rect(fill = "white", color = NA), 
    legend.background = element_rect(fill = "white", color = NA)) +
  geom_text_repel(data = sm_loc, mapping = aes(x=Population_longitude, y=Population_latitude, label=Population_name), 
    nudge_x = .15,  box.padding = 0.5, nudge_y = 1.1, segment.curvature = 0.1, segment.ncp = 3, segment.angle = 20, max.overlaps = 20) 
```
.center[![](https://media1.tenor.com/images/3a0c9909b542717ce9f651d07c4d4592/tenor.gif?itemid=8985245)]

---
# Here is the MAP!!!

```{r fig.width=12, fig.height=7, fig.align = 'center'}
map_plot
```

---
# Final analyses results

# **Calculating genetic differentiation **


```{r, echo = F, include=FALSE}
require(ExploreData)
require(vcfR)
require(tidyverse)
require(reshape2)
require(dplyr)
require(tidyr)
require(adegenet)
require(hierfstat)
require(pegas)
require(mmod)
require(ggplot2)
```

## **Loading the data**
```{r message=FALSE, warning=FALSE, include=TRUE}

vcf <- read.vcfR("chr22.1000g.vcf")
Population_info <- read.table("Population_info.txt", sep="\t", header=TRUE)
```

## **Adjusting the data structure**
  
```{r echo=TRUE, message=FALSE, warning=FALSE, include=TRUE}

tidy <- vcfR2tidy(vcf)
All_samples <- tidy$gt
All_samples = subset(All_samples, select = -c(ChromKey, gt_GT)) # keeping only the needed columns
All_samples$gt_GT_alleles <- gsub("\\|", "", as.character(All_samples$gt_GT_alleles)) # remove the | from the genotype (G|G)
All_samples_filtered <- subset(All_samples, nchar(as.character(gt_GT_alleles)) <= 2) # keep only the SNPs
All_samples_filtered_unique <- subset(All_samples_filtered, POS != "19512392") # remove the position 19512392, because it gives me troubles
All_samples_reshape <- dcast(All_samples_filtered_unique, Indiv ~ POS, value.var = "gt_GT_alleles") # reshape the table from long structure to the wide structure

names(Population_info)[1] <- "Indiv" # Change the column name so that it matches the other data.frame's column name
Mydata <- merge(Population_info, All_samples_reshape, by = "Indiv", all = T) # merge the genotype data with the population info data
```
---
class:left, middle
## **Data conversion**
.center[
```{r echo=TRUE, message=FALSE, warning=FALSE, include=TRUE}

Indiv <- as.character(Mydata$Indiv)

Sex <- as.character(Mydata$Sex)

Superpopulation <- as.character(Mydata$Superpopulation_code)

Population <- as.character(Mydata$Population_name)

locus <- Mydata[, 10:19774]

Mydata1 <- df2genind(locus, ploidy = 2, ind.names = Indiv, pop = Superpopulation, sep = "")
```
]
---
class: middle
## **Observed and expected heterozygosity:** 
  
```{r echo=TRUE, message=FALSE, warning=FALSE, include=TRUE}

Basic_stats <- basic.stats(Mydata1)

boxplot(Basic_stats$perloc)

```
---
class: middle
```{r message=FALSE, warning=FALSE, include=TRUE}

# set the population info into in Mydata1 strata slot.
strata(Mydata1) <- Population_info

```

```{r  message=FALSE, warning=FALSE, include=TRUE}

setPop(Mydata1) <- ~Superpopulation_code 
diff <- diff_stats(Mydata1)

```

```{r  message=FALSE, warning=FALSE, include=FALSE}

per.locus <- melt(diff$per.locus, varnames = c("Locus", "Statistic"))
stats     <- c("Hs", "Ht", "Gst", "Gprime_st", "D", "D")
glob      <- data.frame(Statistic = stats, value = diff$global)

```
---
class: middle
```{r  message=FALSE, warning=FALSE, include=TRUE}

set.seed(20151219) 
reps <- chao_bootstrap(Mydata1, nreps = 10)
summarise_bootstrap(reps, D_Jost) # Using the D_Jost function to summarize.

```
---
class: middle
```{r eval=TRUE, message=FALSE, warning=FALSE, include=TRUE}

distance  <- dist(Mydata1) # Euclidean distance
Strata  <- strata(Mydata1)
Amova <- pegas::amova(distance ~ Superpopulation_code/Population_name, data = Strata, nperm = 0)
```

```{r message=FALSE, warning=FALSE, include=TRUE}

set.seed(20151219)
Amova <- pegas::amova(distance ~ Superpopulation_code/Population_name, data = Strata, nperm = 10)

```
---
```{r message=FALSE, warning=FALSE, include=TRUE}

setPop(Mydata1) <- ~Superpopulation_code
DAPC <- dapc(Mydata1, n.pca = 100, n.da = 4) # 100 principal components; 4 discriminant functions
scatter(DAPC)
```
---

# **Other option to read the .vcf file: using vcfR package **
### Now let's see how to develop packages to analyze genomes.

.center[This package can be used to get summary numbers of a vcf file, including what variants are included and how many of them. It is also possible to count how many times different variant types occur in the genotype data of the vcf file.]


```{rmarkdown Marika Chunk, eval=FALSE, include=FALSE}
---
title: "Summarize vcf data in few functions"
author: Marika Oksanen
date: "'r Sys.Date()'"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{RaukRGenome_vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
```
```{r eval=FALSE, include=FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```
```{r setup}
knitr::opts_chunk$set(eval = FALSE, include = FALSE)
#library(RaukRGenome)
```
---
class: center, middle
![](https://monophy.com/media/8ujkvy3gJ0fGHHb5gx/monophy.gif)

Slides created via the R package [**xaringan**]
.left[(https://github.com/yihui/xaringan)]
.left[(https://spcanelon.github.io/xaringan-basics-and-beyond/slides)]